mono/metadata/object-internals.h

```diff
@@ -574,6 +574,7 @@ struct _MonoInternalThread {
 	MonoThreadHandle *handle;
 	gpointer native_handle;
 	gunichar2  *name;
+	gsize       name_generation;
```

Comment:
Make this `volatile` once monolite finishes.

Reply:
There are same-thread function calls generally in the way, of the compiler knowing it can cache, so *should* be ok either way. Some compilers over-pessimize volatile, at least in some modes (msvc arm modal).